### ISSUE TITLE
Breaks icons in v6.2.0

### ISSUE THREAD

**Issue by ShMcK** [expo/vector-icons#31]
Updating from v6.1.0 to v6.2.0 breaks vector icons.

![screen shot 2017-11-14 at 10 44 56 am](https://user-images.githubusercontent.com/4660659/32798210-e0b1ff38-c928-11e7-8f8f-efd217d60264.png)

I've since rolled back to 6.1.0 without an issue.

**Comment by brentvatne** [comment]
error doesn't really make sense, nothing changed that would impact this. try updating again, should be fine. if not working, try clearing node_modules and installing again

**Comment by brentvatne** [comment]
oh -- are you using babel-preset-expo?

**Comment by ShMcK** [comment]
Yes, with some additional presets.

```js
{
  "presets": [
    "babel-preset-expo",
    "react-native-dotenv"
  ],
  "env": {
    "development": {
      "plugins": [
        "transform-react-jsx-source"
      ]
    }
  }
}
```

**Comment by ShMcK** [comment]
I've tried resetting modules, clearing the npm cache, etc. The only thing that worked was rolling back.

**Comment by brentvatne** [comment]
@ShMcK - any chance you can share your repo? or repro it on another repoistory?

**Comment by ShMcK** [comment]
Unfortunately, its private. I'll try to duplicate the issue in a separate repo.

**Comment by brentvatne** [comment]
thanks!

**Comment by ShMcK** [comment]
I created a smaller repo, but couldn't replicate the problem.

I'm certain the issue exists, at least for me alone, as changing back and forth from v6.2 to v6.1 recreates the issue.

Looking at the library commits, I'm also not sure what may have changed. And following the code within the node_modules, everything in Javascript land checks out

---

According to the "react-native-vector-icons" docs, when facing a similar error message, take the following steps. Unfortunately, none of these actions are available when using Expo.

```
Red screen with "Unrecognized font family" error on iOS

- Make sure you've added manually the reference of your .ttf on your xcodeproj Resources folder.
- Check that the font you are trying to use appears in Info.plist, if you've added the whole folder and it's blue in color, then you need to add it to the path.
- Check that the font is copied in the Copy Bundle Resources in Build Phases.
- Recompile the project.
```

**Comment by brentvatne** [comment]
@ShMcK - if you're unable to repro it on another project, the only thing we could do is have you send me your project and I can investigate. That said, it almost certainly has nothing to do with this library updating because the changes from 6.1 to 6.2 would not have impacted that, and it seems like a coincidence that changing versions causes this problem. There must be something else that is interacting in your project to cause the problem. If you need me to sign a NDA or whatever that's fine.

**Comment by tobowers** [comment]
I have the same problem upgrading from 5.0.0 -> 6.2 I get 'unrecognized font family: material' but same error really.

**Comment by jayeszee** [comment]
Me too! I upgraded from 5.1 to 6.2 and it broke.

**Comment by ShMcK** [comment]
@brentvatne, let me know if you'd like a hands on version to debug. Happy to comply :)

**Comment by jayeszee** [comment]
@brentvatne @ShMcK  found the issue. its this https://github.com/expo/vector-icons/commit/2f496edbbde65c215c404c9b932403c6e6273226 reverting these 4 lines makes it work.

**Comment by brentvatne** [comment]
@ShMcK - haha that would be good! :) maybe grab a coffee and do this tomorrow?

@jayeszee @ShMcK - do you have `babel-preset-expo` in `.babelrc`?

**Comment by jayeszee** [comment]
@brentvatne 
![image](https://user-images.githubusercontent.com/10326244/33453818-da32f95a-d5cb-11e7-989b-543cfd6aeeeb.png)

**Comment by ShMcK** [comment]
I have a similar babelrc. Could do coffee tomorrow, just send me a message on Twitter.

**Comment by jayeszee** [comment]
@brentvatne sorry unfortunately I too am on a private repo =/ I can say that it does work on a fresh new crna app though.

**Comment by jayeszee** [comment]
@brentvatne @ShMcK so i did a bit of digging and it seems like the error is due to the fact that my app which was created a while ago still used 
`AppRegistry.registerComponent` where crna already does for us now. I went over and created a new crna app and just copied some of the initial configurations that differed like exporting my app instead of manually registering the component and now it seems to be happy.

**Comment by brentvatne** [comment]
I see! You should use `Expo.registerRootComponent` -- perhaps we need to give a warning if people register their applications through `AppRegistry.registerComponent` rather than `Expo.registerRootComponent` (which wraps the former in order to get rid of a redundant string argument in the first param and do some initialization work).

**Comment by ShMcK** [comment]
I haven't had the same success with AppRegistry.registerComponent. 

Thanks @jayeszee, I'll try again tomorrow with your recipe of generating a new CRNA app and migrating the configuration over.

**Comment by ShMcK** [comment]
Thanks @brentvatne, for helping resolve my issue.

For anyone else with the same problem, it was a combination of two factors:

1) I was using AppRegistry.registerComponent, which is no longer necessary. With CRNA you can also get away without using any registration call, as it assumes you are building whatever is default exported from "/ROOT/App.js". @jayeszee nailed the problem.

2) In my "app.json", I specified "expo.appKey", which looked for the name of the registered component. Removing this allowed the app to build without a registerComponent named.

**Comment by ShMcK** [comment]
@shamanking, check the capitalization on ionicons. I believe it should be "Ionicons".

**Comment by shamanking** [comment]
Thanks @ShMcK, I do it as @jayeszee and it works. My error arise when I detach expo to the exist project.

**Comment by aleph-naught2tog** [comment]
THANK YOU. I've been trying for ages. The trick was the `registerRootComponent`.

**Comment by tuzmusic** [comment]
I'm pretty sure I started my project with `expo init` and have never needed to register components in any way. But the app won't run on an actual iOS device unless I add `AppRegistry.registerComponent('main',() => App)`. 

As noted, this causes the error with the Fonts (although the error doesn't appear on the device, it just shows no icons; in Expo the error can simply be 

Using `Expo.registerRootComponent` (and `main` key in `package.json`) does indeed fix the Font problem, but then, without the `AppRegistry` line, I'm back to where I started and the app won't load on a device.

Is there a way to fix this?
